lib/cn/blk_list.c

```diff
@@ -50,33 +52,37 @@ commit_mblocks(struct mpool *mp, struct blk_list *blks)
 }
 
 merr_t
-delete_mblock(struct mpool *mp, struct kvs_block *blk)
+delete_mblock(struct mpool *mp, uint64_t mbid)
 {
-    merr_t err = mpool_mblock_delete(mp, blk->bk_blkid);
-    if (err) {
-        log_errx("Failed to delete mblock, blkid 0x%lx", err, blk->bk_blkid);
-        return err;
-    }
+    merr_t err;
 
-    blk->bk_blkid = 0;
+    INVARIANT(mp);
+
+    if (mbid) {
```

Comment:
I would probably prefer an early return here. Doesn't mpool_mblock_delete() know how to handle a 0 mblock id?

Reply:
It does handle `mbid==0`

Reply:
will update